src/context/chartLayoutContext.tsx

```diff
@@ -62,10 +62,48 @@ export const useOffset = (): ChartOffsetRequired => {
   return useAppSelector(selectChartOffset) ?? manyComponentsThrowErrorsIfOffsetIsUndefined;
 };
 
+/**
+ * Returns the width of the chart in pixels.
+ *
+ * If you are using chart with hardcoded `width` prop, then the width returned will be the same
+ * as the `width` prop on the main chart element.
+ *
+ * If you are using a chart with a `ResponsiveContainer`, the width will be the size of the chart
+ * as the ResponsiveContainer has decided it would be.
+ *
+ * If the chart has any axes or legend, the `width` will be the size of the chart
+ * including the axes and legend. Meaning: adding axes and legend will not change the width.
+ *
+ * The dimensions do not scale, meaning as user zoom in and out, the width number will not change
+ * as the chart gets visually larger or smaller.
+ *
+ * Returns `undefined` if used outside a chart context.
+ *
+ * @returns {number | undefined} The width of the chart in pixels, or `undefined` if not in a chart context.
+ */
```

Comment:
I think these are fine but is a combined "useChartDimentions" nicer to consume? 1 hook call and two properties rather than 2 hook calls?

Not a big deal but curious

Reply:
I don't have a strong preference. If it's two different hooks then we avoid the memoization discussion. But that isn't a difficult problem to solve either.

Reply:
Eh let's ship it like this. If there's someone in the aether who cares we can accommodate later but access to the things is what most care about, no matter how